packages/kit/src/utils.ts

```diff
@@ -12,7 +12,10 @@ export function toArray<T> (value: T | T[]): T[] {
 export function filterInPlace<T> (array: T[], predicate: (item: T, index: number, arr: T[]) => unknown) {
   for (let i = array.length; i--; i >= 0) {
     if (!predicate(array[i]!, i, array)) {
-      array.splice(i, 1)
+      const last = --array.length
+      if (i < last) {
+        array[i] = array[last]!
```

Comment:
But this changes the order of the array, I am not sure this micro performance improvement would be worth it.